packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -173,19 +240,93 @@ class RawMenuAnchor extends StatefulWidget {
     this.consumeOutsideTaps = false,
     this.onOpen,
     this.onClose,
+    this.onOpenRequested = _defaultOnOpenRequested,
+    this.onCloseRequested = _defaultOnCloseRequested,
     this.useRootOverlay = false,
     this.builder,
     required this.controller,
     required this.overlayBuilder,
     this.child,
   });
 
-  /// A callback that is invoked when the menu is opened.
+  /// Called when the menu overlay is shown.
+  ///
+  /// This callback is triggered when the menu overlay is inserted into the
+  /// widget tree, typically before any opening animations begin. A typical
+  /// usage is to respond when the menu first becomes interactive, such as by
+  /// setting focus to a menu item.
+  ///
+  /// This callback is called regardless of whether the menu is already open or
+  /// the menu's position has changed.
   final VoidCallback? onOpen;
 
-  /// A callback that is invoked when the menu is closed.
+  /// Called when the menu overlay is hidden.
+  ///
+  /// This callback is triggered when the menu overlay is removed from the
+  /// widget tree, typically after any closing animations have completed.
+  ///
+  /// If a callback is provided for [onCloseRequested], [onClose] will not be
+  /// called until after `hideOverlay` is called.
   final VoidCallback? onClose;
 
+  /// Called when a request is made to open the menu.
+  ///
+  /// This callback is used by themed menu widgets to intercept open requests,
+  /// typically with the intent of adding a delay or an animation.
+  ///
+  /// After an open request is intercepted, the handler is responsible for
+  /// eventually calling `showOverlay`. If needed, the call can be made after a
+  /// delay, or not at all. The `showOverlay` call sets [MenuController.isOpen]
+  /// to true, builds (or rebuilds) the [RawMenuAnchor] its overlay (the widget
+  /// built by [overlayBuilder]), and ensures that the menu overlay is showing.
+  /// If a [RawMenuAnchor] is used in a themed menu that plays an opening
+  /// animation, `showOverlay` should come before the opening animation, since
+  /// the animation plays on the overlay itself.
+  ///
+  /// The `position` argument is the `position` that [MenuController.open] was
+  /// called with. When a menu is repositioned, [onOpenRequested] may be called
+  /// with a new `position` value while the menu is still open.
```

Comment:
Moving this sentence to the top, where I think fits better. Instead of emphasizing whether the position could be new, it's more important that the callback will always be called regardless of whether the position is different or whether the overlay is already showing.
```suggestion
  /// called with.
```

Reply:
Sorta done. Check the paragraph starting on line 286.